apps/evm/lib/evm/message_call.ex

```diff
@@ -199,7 +199,7 @@ defmodule EVM.MessageCall do
           %{machine_state | last_return_data: output}
 
         true ->
-          machine_state = Memory.write(machine_state, out_offset, output)
+          machine_state = Memory.write(machine_state, out_offset, output, out_size)
```

Comment:
Is there a test that we could add for this? It seems like this was an important piece that was overlooked, and in terms of making sure we don't re-introduce this bug in the future, it would be nice to have a test that covers this. What do you think?

Reply:
@germsvel done